src/borrow_check/type_check.md

```diff
@@ -8,3 +8,59 @@ this type-check, we also uncover the region constraints that apply to
 the program.
 
 TODO -- elaborate further? Maybe? :)
```

Comment:
Maybe delete this TODO?

Reply:
I think that todo is still pretty relevant, as dealing with user types is only a small part of how the mir typechecker works